hypershift-operator/main.go

```diff
@@ -217,6 +218,7 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 		EnableOCPClusterMonitoring: opts.EnableOCPClusterMonitoring,
 		EnableCIDebugOutput:        opts.EnableCIDebugOutput,
 		ImageMetadataProvider:      &util.RegistryClientImageMetadataProvider{},
+		MetricsSet:                 metrics.MetricsSetFromEnv(),
```

Comment:
right now this is unsanitized user input, as HO consumer if I make a typo in env variable, that would fail silently and the default would be used.

Can we instead let MetricsSetFromEnv (or have an additional validate function) to validate the input and fail with a meaningful message if none of the allowed values or empty is passed?